### DISCUSSION TITLE
Failed to create record: DrySubmit create rule failure: sql: no rows in result set

### DISCUSSION THREAD

**Discussion by elliotsoomro** [pocketbase/pocketbase#1200] [Q&A] [answered]
Just updated to Pocketbase v0.9.0 and getting this error when trying to create a record. Only seem to affect this collection. Recently deleted all rows using sqlite3 CLI.

![Screenshot 2022-12-07 at 16 04 25](https://user-images.githubusercontent.com/10349178/206214510-df7e0db5-8219-4563-ad5d-6a13a79acadb.jpg)
![Screenshot 2022-12-07 at 16 02 24 1](https://user-images.githubusercontent.com/10349178/206214636-2eac806f-9a6b-4055-8d3e-064691f2fd13.jpg)
![Screenshot 2022-12-07 at 16 01 58](https://user-images.githubusercontent.com/10349178/206214092-d7209bb8-ee28-4915-b179-3dd7f1053eb6.jpg)

![Screenshot 2022-12-07 at 16 01 25](https://user-images.githubusercontent.com/10349178/206214066-3de1a7dc-cff9-479b-9dba-9123aabbbd72.jpg)
![Screenshot 2022-12-07 at 16 01 37](https://user-images.githubusercontent.com/10349178/206214084-4787c4ff-c58a-49fe-bf45-c1ad99afd4ec.jpg)

**Comment by ganigeorgiev** [comment, marked as answer]
From the screenshot, the "Create API rule" requires valid app user authentication in order to perform the create request.

Could you try authenticating before the create call and try again?

**Reply by elliotsoomro** [reply]
Yes, tried logging out and in again. Also added 

`await pocketbase.collection('users').authWithPassword('<redacted>', '<redacted>')`

right before the API call. Still the same error.

**Reply by ganigeorgiev** [reply]
Actually, on second look at the logs screenshot it seems that the request was authorized so the error could be from something else.

Could you try starting the PocketBase executable with `--debug` and check the logs in the console?

**Reply by elliotsoomro** [reply]
Like this?

![Screenshot 2022-12-07 at 16 30 14](https://user-images.githubusercontent.com/10349178/206221039-ee7e4933-8fb8-435c-bf91-6b80eb11dcee.jpg)

**Reply by ganigeorgiev** [reply]
Yes, but I just noticed that you are probably missing `!` in the Create API rule and it practically allows the request to be performed only by admins OR guests, aka. your rule probably should be:

```
@request.auth.id != ""
```

**Reply by elliotsoomro** [reply]
Of course! That was a silly mistake. Sorry for the inconvenience. Thanks for your quick response!

**Reply by Mawulijo** [reply]
Hello @ganigeorgiev 
Below is my code all-in-one code for auth, used to work fine but does not work anymore. I get `Failed create rule constraint: sql: no rows in result set` error and `Error code 400` in the catch block. I have not made any updates to the users table and I realized when I manually create a user with the expected email manually before trying the oAuth2 it works

I'm using `v0.16.7`

<img width="815" alt="Screenshot 2023-08-02 at 3 41 18 PM" src="https://github.com/pocketbase/pocketbase/assets/15157923/c08e0070-b46b-4df2-905c-740cdb53aa79">

```async function signIn() {
		await pb
			.collection('users')
			.authWithOAuth2({
				provider: 'google',
				createData: {
					department: '',
					emailVisibility: true,
					enrollment_year: '',
					faculty: '',
					level: '',
					name: '',
					program: '',
					role: [],
					session: '',
					status: '',
					username: '',
				}
			})
			.then(async (result) => {
				console.log('Result:', result.record.email);
				const identity = await pb
					.collection('identities')
					.getFirstListItem(`email = '${result.record.email}'`);
				if (result.record.username.includes('users') && identity.role === '') {
					window.location.href = `profile`;
				} else if (result.record.username.includes('users')) {
					window.location.href = `profile/setrole`;
			})
			.catch(async (error) => {
				console.log(pb.authStore.isValid);
				console.log(pb.authStore.token);
				console.log(pb.authStore.model?.id);
				let message = errorMessage(error);
				console.log('Error:', message);
			});
		}```

**Reply by Mawulijo** [reply]
<img width="664" alt="Screenshot 2023-08-02 at 3 47 32 PM" src="https://github.com/pocketbase/pocketbase/assets/15157923/577d12ee-f7af-462a-87e4-e020d1760003">

**Reply by ganigeorgiev** [reply]
@Mawulijo The error indicates that the OAuth2 sign-up request didn't satisfy your "users" collection Create API rule.

Why do you have `id = @request.auth.id` in the first place for the create rule?

**Reply by Mawulijo** [reply]
Wow, very silly of me. I actually don't remember why I put it there but I may me trying to prevent just about any one from creating records
It works now after removing it

I appreciate your rapid response and do you have any suggested rule aside leaving it empty for new signups?

**Reply by ganigeorgiev** [reply]
>  do you have any suggested rule aside leaving it empty for new signups?

If you don't need any additional request data validations, then it is safe to leave the Create rule empty.
